### ISSUE TITLE
Router.start() called more than once

### ISSUE THREAD

**Issue by wallzero** [ui-router/react#606]
With `@uirouter/react@1.0.1`, I receive the following error with `<React.StrictMode>`:

```
core.js:71 Uncaught Error: 
  The Router.start() method has been called more than once.

  The <UIRouter> component calls start() as final step of the initialization and you shouldn't need to call it manually.
```

**Comment by wallzero** [comment]
The reason is with `<UIRouter>` having been switched to a function component, `useRef` needs to be wrapped in a `useEffect`. Otherwise multiple renders can trigger before `useRef.current` is set.

**Comment by christopherthielen** [comment]
fixed by #607
